### DISCUSSION TITLE
SelectOneMenu not showing options after PrimeFaces upgrade

### DISCUSSION THREAD

**Discussion by steffenscheffler** [primefaces/community#2129] [PrimeFaces] [answered]
Hey :) I have updated PrimeFaces from Version 8 to 13 in a bigger project using the migration guides. Most things went pretty well - but all of the "p:selectOneMenu" do no longer work (>100 occurences). They show no value at all; and when clicking on them, nothing shows up in the dropdown list. As an example, here is one of the components:

```
<p:selectOneMenu id="filter_sent"
                                             value="#{filterBean.filterSent}">
    <f:selectItem itemLabel="#{msg['filter.all']}" itemValue="#{null}" noSelectionOption="true"/>
    <f:selectItem itemLabel="#{msg['filter.today']}" itemValue="1"/>
    <p:ajax event="change"
            listener="#{classAction.onValidateFilter}"
            update="activeFilter"/>
</p:selectOneMenu>
```

When I add var and p:column, then a value is shown:

```
<p:selectOneMenu id="filter_sent" var="filter"
                                             value="#{filterBean.filterSent}">
    <f:selectItem itemLabel="#{msg['filter.all']}" itemValue="#{null}" noSelectionOption="true"/>
    <f:selectItem itemLabel="#{msg['filter.today']}" itemValue="1"/>
    <p:column>#{filter}</p:column>
    <p:ajax event="change"
            listener="#{classAction.onValidateFilter}"
            update="activeFilter"/>
</p:selectOneMenu>
```

But unfortunately this does not show the actual label, but it will show the itemValue, e.g. "1" in the dropdown list. After selecting it, the correct label value is shown for the then selected value.
I think this might be related to CSS changes after the Upgrade, but unfortunately I am really inexperienced with CSS and have no idea where to start :-) Could anybody give me a heads up on what changed between the versions? (A lot of other css is "broken" aswell, but things remain usable - they just don't look good, but this will be taken care of later by the styling team. I just have to make sure that everything works functional-wise before handing over to them)
Thank you very much in advance!

**Comment by ks-pang-hao** [comment]
1. Make sure your #{filterBean.filterSent} is the String type. otherwise, you have to specifiy the data type with the attributre of [converter].
2. Remove all custom css from header then to try if the component works correctly to prevent the css errors.
3. Open the development mode from your chrome and obseve any error message from the console tab.

Maybe it helps.

**Reply by melloware** [reply]
agreed the SelectOne you should not need to add the var and the column it should just work.  I feel like @ks-pang-hao mentioned you are missing something.  Compare with the PrimeFaces showcase.

Are you using a homegrown theme or a PrimeFaces Premium Template?

**Reply by steffenscheffler** [reply]
Thank you very much for the input! The exact same code (first example) worked without any issues with primefaces 8 and also compared to the showcase it looks correct to me. Only after the upgrade to version 13, the menu stays empty (and does not show anything when clicking on it). As I mentioned, I feel like it is somehow related to CSS maybe, as a lot of our CSS broke after the upgrades. We are using an own theme, maybe this is part of the problem aswell. But as our themes and css is quite big and I am lacking knowledge in that area, I am not sure where to look ;-) I could do the workaround with p:column after every selectOneMenu, it does work, but it seems to be not intended that way.
Also, no errors on browser console.

**Reply by melloware** [reply]
ok then yeah it might be CSS related.  Because I can't think of any other reason it would not work.

**Reply by steffenscheffler** [reply]
@melloware  Thank you. I have removed custom css now, but the problem still persists. Maybe I have done something wrong with the upgrade - I have updated the PrimeFaces version in gradle to 13.0.10. Afterwards I fixed all compilation issues according to the migration guides. 

In DOM it seems like there is a `<div class="ui-helper-hidden-accessible">` that contains all the options that I would expect (selectItems). If I remove that css class, it shows a dropdown box with all values right above the actual primeFaces selectOneMenu

**Reply by melloware** [reply, marked as answer]
Yes that hidden needs to stay hidden that is what holds the form value for submit while the UI renders a different styled component.

@steffenscheffler start simple with a  an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.  Just put a simple SelectOne in that Test project with PF 13.0.10 and see if it works.  If it does then you are back to there is something wrong with your code hiding the select one.

**Reply by tandraschko** [reply]
also maybe try 14.0.1 ;D

**Reply by vieira-brz** [reply]
Here's a breakdown of how to debug and fix this:

**1. Isolate the issue:**

* **Create a minimal example:** As suggested, start with a simple PrimeFaces Test project. This will help you isolate if the issue is specific to your project or a broader PrimeFaces issue.
* **Remove custom CSS:** Temporarily disable your custom stylesheets (or comment out the relevant sections). If the `p:selectOneMenu` starts working, you know the issue is in your CSS.
* **Check browser console:** Look for any JavaScript errors or CSS conflicts in the browser console. This might provide clues about the problem.

**2. Troubleshoot CSS Conflicts:**

* **PrimeFaces default styles:** PrimeFaces 13.0.10 uses a different CSS structure than earlier versions. Ensure that you aren't overriding PrimeFaces' default stylesheets with your custom CSS.  
* **Specificity:** CSS conflicts can happen due to overriding styles based on specificity (element selectors, IDs, and classes).  Look for selectors in your custom CSS that might be affecting the `ui-helper-hidden-accessible` div.
* **Inspect Element:** Use your browser's developer tools to inspect the `p:selectOneMenu` element. Look for any CSS styles that are being applied to the `ui-helper-hidden-accessible` div that might be preventing it from being displayed. 

**3. Potential CSS Solutions:**

* **Add an exception in your CSS:** You can use the `!important` declaration to override conflicting styles:
   ```css
   .ui-helper-hidden-accessible {
       display: none !important; /* PrimeFaces default */
   }
   #your-select-one-menu-id .ui-helper-hidden-accessible {
       display: block !important; /* Override for your specific menu */
   }
   ```
* **Use `!important` sparingly:** While it can be a quick fix, using `!important` too often can lead to CSS maintenance issues. It's better to understand the specificity issue and resolve it more elegantly.
* **Use a different CSS approach:** If you have a complex theme, consider separating your PrimeFaces CSS overrides into a dedicated stylesheet.

**4. Further Debugging Tips:**

* **Enable PrimeFaces debugging:**  Add this to your `web.xml`:
   ```xml
   <context-param>
       <param-name>primefaces.DEBUG</param-name>
       <param-value>true</param-value>
   </context-param>
   ```
* **Inspect the rendered HTML:** Make sure that the `p:selectOneMenu` is being rendered correctly in the HTML. Look for missing attributes or mismatched HTML structure.

**Reply by kurtomerfaruk** [reply]
I was encountering similar problems, but I was using Jakarta Faces 4.0.4. So it was not about primefaces, it was about jakarta-faces-4.0.4. Could you have the same problem?

**Reply by melloware** [reply]
Closing this as inactive. Will re-open if reproducer is provided.
